### PR TITLE
MTDSA-5579 create alignment

### PR DESCRIPTION
**Pull request by zachhall1234**

### CODE REVIEW

File: app/v1/models/errors/mtdErrors.scala [hunk 1]
```diff
@@ -79,6 +80,21 @@ object RuleToDateError
 object RuleNoChangeError
   extends MtdError("RULE_NO_CHANGE", "The deducted amount has not changed")
 
+object RuleUnalignedDeductionPeriodError
+  extends MtdError("RULE_UNALIGNED_DEDUCTIONS_PERIOD", "The deductions periods do not align with the tax year supplied")
+
+object RuleDeductionDateRangeError
```
Comment (ian-ralphs) [inline app/v1/models/errors/mtdErrors.scala:86]: I think this is a duplicate of an existing error.

File: app/v1/services/CreateService.scala [hunk 2]
```diff
@@ -47,15 +46,16 @@ class CreateService @Inject()(connector: CreateConnector) extends DesResponseMap
...
-      "INVALID_DATE_FROM" -> FromDateFormatError,
-      "INVALID_DATE_TO" -> ToDateFormatError,
-      "INVALID_DEDUCTIONS_DATE_RANGE" -> RuleDeductionsDateRangeInvalidError,
-      "INVALID_DEDUCTIONS_TO_DATE_BEFORE_DEDUCTIONS_FROM_DATE" -> RuleToDateBeforeFromDateError,
-      "NOT_FOUND" -> NotFoundError,
+      "INVALID_TAXABLE_ENTITY_ID" -> NinoFormatError,
+      "INVALID_PAYLOAD" -> RuleIncorrectOrEmptyBodyError,
+      "INVALID_EMPREF" -> EmployerRefFormatError,
+      "INVALID_REQUEST_TAX_YEAR_ALIGN" -> RuleUnalignedDeductionPeriodError,
+      "INVALID_REQUEST_DATE_RANGE" -> RuleDeductionDateRangeError,
```
Comment (ian-ralphs) [inline app/v1/services/CreateService.scala:53]: RuleDeductionDateRangeInvalid

File: app/v1/controllers/requestParsers/validators/CreateValidator.scala [hunk 3]
```diff
@@ -51,7 +51,8 @@ class CreateValidator extends Validator[CreateRawData] with FixedConfig {
       PeriodDataDeductionDateValidation.validate(data.body, "deductionToDate", DeductionToDateFormatError),
       DateValidation.validate(FromDateFormatError)(req.fromDate),
       DateValidation.validate(ToDateFormatError)(req.toDate),
-      ToBeforeFromDateValidation.validate(req.fromDate, req.toDate, RuleDateRangeInvalidError)
+      ToBeforeFromDateValidation.validate(req.fromDate, req.toDate, RuleDateRangeInvalidError),
```
[the remark this answers is not in the document]
Comment (zachhall1234) [reply app/v1/controllers/requestParsers/validators/CreateValidator.scala]: Removed the the ToBeforeFromDateValidation

File: app/v1/models/errors/mtdErrors.scala (hunk 1 above)
Comment (zachhall1234) [reply app/v1/models/errors/mtdErrors.scala]: Removed duplicate and used what was previously there

File: app/v1/services/CreateService.scala (hunk 2 above)
Comment (zachhall1234) [reply app/v1/services/CreateService.scala]: Changed here & specs

**Pull request merged**